core/src/main/kotlin/com/justai/jaicf/model/activation/ActivationRule.kt

```diff
@@ -50,4 +50,9 @@ abstract class ActivationRuleAdapter : ActivationRule {
         onlyIf.add(predicate)
         return this
     }
+
+    fun getOnlyIfPredicates() = onlyIf
+
+    fun setOnlyIfPredicates(predicates: List<ActivationRule.OnlyIfContext.() -> Boolean>) =
+        onlyIf.addAll(predicates)
```

Comment:
Using `add` function in a `set` method might be confusing. I'd suggest to use a single verb and logic in a method's name and method's code. Either leave it as `set`, but change the code to `onlyIf = predicates`, or rename the method to `add`.
If you choose `set` logic, it might be implemented as a simple `var` property